### PR TITLE
8.0 t003583 new

### PR DESCRIPTION
**Pull request by azmimr67**

### CODE REVIEW

File: accountant_general_audit_index_a1104/README.rst [hunk 1]
```diff
@@ -0,0 +1,48 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+====================================
+General Audit Index A1104
```
Comment (andhit-r) [inline accountant_general_audit_index_a1104/README.rst]: A.110.4

File: accountant_general_audit_index_a1104/README.rst [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+Installation
+============
+
+To install this module, you need to:
+
+1.  Clone the branch 8.0 of the repository https://github.com/open-synergy/opnsynid-vertical-accountant
+2.  Add the path to this repository in your configuration (addons-path)
+3.  Update the module list
+4.  Go to menu *Setting -> Modules -> Local Modules*
+5.  Search For *General Audit Index A1104*
```
Comment (andhit-r) [inline accountant_general_audit_index_a1104/README.rst]: A.110.4

File: accountant_general_audit_index_a1104/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
+# Copyright 2021 OpenSynergy Indonesia
+# Copyright 2021 PT. Simetri Sinergi Indonesia
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+# pylint: disable=locally-disabled, manifest-required-author/
+
+{
+    "name": "General Audit Index A1104",
```
Comment (andhit-r) [inline accountant_general_audit_index_a1104/__openerp__.py]: A.110.4

File: accountant_general_audit_index_a1104/data/base_sequence_configurator_data.xml [hunk 4]
```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2020 OpenSynergy Indonesia
+     Copyright 2020 PT. Simetri Sinergi Indonesia
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+<openerp>
+<data>
+    <record
+            id="base_sequence_accountant_general_audit_index_a1104"
```
Comment (andhit-r) [inline accountant_general_audit_index_a1104/data/base_sequence_configurator_data.xml]: XML ID tidak boleh sama dengan XML ID sequence

File: accountant_general_audit_index_a1104/models/accountant_general_audit_index_a1104.py [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+# Copyright 2021 PT. Simetri Sinergi Indonesia
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import _, api, fields, models
+from openerp.exceptions import Warning as UserError
+
+
+class AccountantGeneralAuditIndexA1104(models.Model):
+    _name = "accountant.general_audit_index_a1104"
+    _description = "General Audit Index A1104"
```
Comment (andhit-r) [inline accountant_general_audit_index_a1104/models/accountant_general_audit_index_a1104.py]: A.110.4

File: accountant_general_audit_index_a1104/models/accountant_general_audit_index_a1104.py [hunk 6]
```diff
@@ -0,0 +1,306 @@
...
+    _state_to = ["valid"]
+
+    @api.multi
+    @api.depends(
+        "company_id",
+    )
+    def _compute_policy(self):
+        _super = super(AccountantGeneralAuditIndexA1104, self)
+        _super._compute_policy()
+
```
Comment (andhit-r) [inline accountant_general_audit_index_a1104/models/accountant_general_audit_index_a1104.py:30]: Tambahkan method **_compute_general_audit_id**. Lihat contoh di https://github.com/open-synergy/opnsynid-vertical-accountant/blob/423736d889e47992957e785851863937fc9b657c/accountant_general_audit_index_a1102/models/accountant_general_audit_index_a1102.py#L35

Tambahkan method **_inverse_general_audit**. Contoh lihat di https://github.com/open-synergy/opnsynid-vertical-accountant/blob/423736d889e47992957e785851863937fc9b657c/accountant_general_audit_index_a1102/models/accountant_general_audit_index_a1102.py#L45 .

File: accountant_general_audit_index_a1104/models/accountant_general_audit_index_a1104.py [hunk 7]
```diff
@@ -0,0 +1,306 @@
...
+    general_audit_id = fields.Many2one(
+        string="General Audit",
+        comodel_name="accountant.general_audit",
+        required=True,
+        states={
+            "draft": [
+                ("readonly", False),
+            ],
+        },
+    )
```
Comment (andhit-r) [inline accountant_general_audit_index_a1104/models/accountant_general_audit_index_a1104.py]: Sesuaikan **general_audit_id** sesuai dengan https://github.com/open-synergy/opnsynid-vertical-accountant/blob/423736d889e47992957e785851863937fc9b657c/accountant_general_audit_index_a1102/models/accountant_general_audit_index_a1102.py#L63-L75

Tambahkan **general_audit_ids**. Lihat contoh di https://github.com/open-synergy/opnsynid-vertical-accountant/blob/423736d889e47992957e785851863937fc9b657c/accountant_general_audit_index_a1102/models/accountant_general_audit_index_a1102.py#L76-L82. Sesuaikan **relation** dan **column1**

File: accountant_general_audit_index_a1104/models/accountant_general_audit_index_a1104.py [hunk 8]
```diff
@@ -0,0 +1,306 @@
...
+        string="Accoount Type Set",
+        comodel_name="accountant.client_account_type_set",
+        related="general_audit_id.account_type_set_id",
+        readonly=True,
+        store=True,
+    )
+    company_id = fields.Many2one(
+        string="Company",
+        comodel_name="res.company",
+        required=True,
```
Comment (andhit-r) [inline accountant_general_audit_index_a1104/models/accountant_general_audit_index_a1104.py:139]: Buang

File: accountant_general_audit_index_a1104/models/accountant_general_audit_index_a1104.py [hunk 9]
```diff
@@ -0,0 +1,306 @@
...
+        readonly=True,
+        store=True,
+        related="general_audit_id.company_id",
+    )
+    partner_id = fields.Many2one(
+        string="Partner",
+        comodel_name="res.partner",
+        related="general_audit_id.partner_id",
+        required=False,
```
Comment (andhit-r) [inline accountant_general_audit_index_a1104/models/accountant_general_audit_index_a1104.py:147]: Buang

File: accountant_general_audit_index_a1104/models/accountant_general_audit_index_a1104.py [hunk 10]
```diff
@@ -0,0 +1,306 @@
...
+        readonly=True,
+        store=True,
+    )
```
Comment (andhit-r) [inline accountant_general_audit_index_a1104/models/accountant_general_audit_index_a1104.py:150]: Tambahkan https://github.com/open-synergy/opnsynid-vertical-accountant/blob/423736d889e47992957e785851863937fc9b657c/accountant_general_audit_index_a1102/models/accountant_general_audit_index_a1102.py#L103-L148

File: accountant_general_audit_index_a1104/models/accountant_general_audit_index_a1104.py [hunk 11]
```diff
@@ -0,0 +1,306 @@
...
+    status_id = fields.Many2one(
+        string="Status",
+        comodel_name="accountant.general_audit_index_a1104_status",
+        required=True,
+        states={
+            "draft": [
+                ("readonly", False),
+            ],
+        },
+    )
```
Comment (andhit-r) [inline accountant_general_audit_index_a1104/models/accountant_general_audit_index_a1104.py:216]: Tambahkan https://github.com/open-synergy/opnsynid-vertical-accountant/blob/423736d889e47992957e785851863937fc9b657c/accountant_general_audit_index_a1102/models/accountant_general_audit_index_a1102.py#L285-L292

File: accountant_general_audit_index_a1104/models/accountant_general_audit_index_a1104_status.py [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+# Copyright 2021 OpenSynergy Indonesia
+# Copyright 2021 PT. Simetri Sinergi Indonesia
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import fields, models
+
+
+class AccountantGeneralAuditIndexA1104Status(models.Model):
+    _name = "accountant.general_audit_index_a1104_status"
+    _description = "Index A1104 Status"
```
Comment (andhit-r) [inline accountant_general_audit_index_a1104/models/accountant_general_audit_index_a1104_status.py]: A.110.4

File: accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_status_views.xml [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+        <field name="name">Index A1104 Status</field>
+        <field name="type">ir.actions.act_window</field>
+        <field name="res_model">accountant.general_audit_index_a1104_status</field>
+        <field name="view_type">form</field>
+        <field name="view_mode">tree,form</field>
+    </record>
+
+    <menuitem
+            id="accountant_general_audit_index_a1104_status_menu"
+            name="Index A1104 Status"
```
Comment (andhit-r) [inline accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_status_views.xml]: A.110.4

File: accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_status_views.xml [hunk 14]
```diff
@@ -0,0 +1,90 @@
...
+            parent="accountant_general_audit.accountant_configuration_general_audit_menu"
```
Comment (andhit-r) [inline accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_status_views.xml:83]: accountant_general_audit.accountant_configuration_general_audit_menu

File: accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml [hunk 15]
```diff
@@ -0,0 +1,314 @@
...
+                    <filter
+                            name="grp_general_audit"
+                            string="General Audit"
+                            context="{'group_by':'general_audit_id'}"
+                        />
+                    <filter
+                            name="grp_state"
+                            string="State"
+                            context="{'group_by':'state'}"
+                        />
```
Comment (andhit-r) [inline accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml:76]: Tambahkan https://github.com/open-synergy/opnsynid-vertical-accountant/blob/423736d889e47992957e785851863937fc9b657c/accountant_general_audit_index_a1102/views/accountant_general_audit_index_a1102_views.xml#L78-L115

File: accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml [hunk 16]
```diff
@@ -0,0 +1,314 @@
...
+            </search>
+        </field>
+    </record>
+
+    <record id="accountant_general_audit_index_a1104_view_tree" model="ir.ui.view">
+        <field name="name">accountant.general_audit_index_a1104 tree</field>
+        <field name="model">accountant.general_audit_index_a1104</field>
+        <field name="arch" type="xml">
+            <tree
+                    fonts="bold:message_unread==True"
```
Comment (andhit-r) [inline accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml]: Hapus krn obj tidak depend ke ir.need.action

File: accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml [hunk 17]
```diff
@@ -0,0 +1,314 @@
...
+                    colors="red:(state in ('confirm','valid'));blue:state == 'draft';grey:state == 'cancel';"
+                >
+                <field name="message_unread" invisible="1" />
```
Comment (andhit-r) [inline accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml]: Hapus krn obj tidak depend ke ir.need.action

File: accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml [hunk 18]
```diff
@@ -0,0 +1,314 @@
...
+                            <field name="account_type_set_id" />
+                            <field
+                                    name="company_id"
+                                    groups="base.group_multi_company"
+                                />
+                            <field
+                                    name="partner_id"
+                                    domain="[('parent_id','=',False)]"
+                                />
+                        </group>
```
Comment (andhit-r) [inline accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml]: Sesuaikan dengan https://github.com/open-synergy/opnsynid-vertical-accountant/blob/423736d889e47992957e785851863937fc9b657c/accountant_general_audit_index_a1102/views/accountant_general_audit_index_a1102_views.xml#L250-L277

File: accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml [hunk 19]
```diff
@@ -0,0 +1,314 @@
...
+                            <group name="grup_question" colspan="4" col="2">
+                                <group name="question_1" colspan="4" col="2">
+                                    <field name="question_1" />
+                                    <field name="question_2" />
+                                    <field name="question_3a" />
+                                    <field name="question_3b" />
+                                </group>
+                                <group name="stat_question" colspan="4" col="2">
+                                    <field name="status_id" />
+                                </group>
```
Comment (andhit-r) [inline accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml]: Sesuaikan dengan https://github.com/open-synergy/opnsynid-vertical-accountant/blob/423736d889e47992957e785851863937fc9b657c/accountant_general_audit_index_a1102/views/accountant_general_audit_index_a1102_views.xml#L318-L321

File: accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml [hunk 20]
```diff
@@ -0,0 +1,314 @@
...
+                </div>
+            </form>
+        </field>
+    </record>
+
+    <record
+            id="accountant_general_audit_index_a1104_action"
+            model="ir.actions.act_window"
+        >
+        <field name="name">General Audit Index A1104</field>
```
Comment (andhit-r) [inline accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml]: A.110.4

File: accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml [hunk 21]
```diff
@@ -0,0 +1,314 @@
...
+            'readonly_by_pass': True,
+            'search_default_dom_draft':1,
+            'search_default_dom_confirm':1,
+            }
+        </field>
+    </record>
+
+    <menuitem
+            id="accountant_general_audit_index_a1104_menu"
+            name="Index A1104"
```
Comment (andhit-r) [inline accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml]: A.110.4

**Review by andhit-r — CHANGES_REQUESTED** [review]
Jangan lupa tambahkan:
1. accountant_general_audit.py https://github.com/open-synergy/opnsynid-vertical-accountant/blob/8.0/accountant_general_audit_index_a1102/models/accountant_general_audit.py . Sesuaikan isinya
2. accountant_general_audit_views.xml https://github.com/open-synergy/opnsynid-vertical-accountant/blob/8.0/accountant_general_audit_index_a1102/views/accountant_general_audit_views.xml

File: accountant_general_audit_index_a1104/README.rst (hunk 1 above)
Comment (azmimr67) [reply accountant_general_audit_index_a1104/README.rst]: done

File: accountant_general_audit_index_a1104/README.rst (hunk 2 above)
Comment (azmimr67) [reply accountant_general_audit_index_a1104/README.rst]: done

File: accountant_general_audit_index_a1104/__openerp__.py (hunk 3 above)
Comment (azmimr67) [reply accountant_general_audit_index_a1104/__openerp__.py]: done

File: accountant_general_audit_index_a1104/data/base_sequence_configurator_data.xml (hunk 4 above)
Comment (azmimr67) [reply accountant_general_audit_index_a1104/data/base_sequence_configurator_data.xml]: done

File: accountant_general_audit_index_a1104/models/accountant_general_audit_index_a1104.py (hunk 5 above)
Comment (azmimr67) [reply accountant_general_audit_index_a1104/models/accountant_general_audit_index_a1104.py]: done

File: accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml (hunk 17 above)
Comment (azmimr67) [reply accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml]: done

File: accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml (hunk 18 above)
Comment (azmimr67) [reply accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml]: done

File: accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml (hunk 19 above)
Comment (azmimr67) [reply accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml]: done

File: accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml (hunk 20 above)
Comment (azmimr67) [reply accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml]: done

File: accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml (hunk 21 above)
Comment (azmimr67) [reply accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml]: done

File: accountant_general_audit_index_a1104/models/accountant_general_audit_index_a1104.py (hunk 10 above)
Comment (andhit-r) [reply accountant_general_audit_index_a1104/models/accountant_general_audit_index_a1104.py:150]: @azmimr67 , ini masih ketinggalan

File: accountant_general_audit_index_a1104/models/accountant_general_audit_index_a1104_status.py (hunk 12 above)
Comment (andhit-r) [reply accountant_general_audit_index_a1104/models/accountant_general_audit_index_a1104_status.py]: Masih ketinggalan

File: accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml (hunk 15 above)
Comment (andhit-r) [reply accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml:76]: Masih ketinggalan

File: accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml [hunk 22]
```diff
@@ -0,0 +1,317 @@
...
+                            <field
+                                    name="company_id"
+                                    groups="base.group_multi_company"
+                                />
+                            <field
+                                    name="partner_id"
+                                    domain="[('parent_id','=',False)]"
+                                />
+                        </group>
+                    </group>
```
Comment (andhit-r) [inline accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml]: Samakan dengan https://github.com/open-synergy/opnsynid-vertical-accountant/blob/08dbedbc02d973c7af3d92897cf1dc47f49dfa0c/accountant_general_audit_index_a1101/views/accountant_general_audit_index_a1101_views.xml#L250-L277

File: accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml [hunk 23]
```diff
@@ -0,0 +1,317 @@
...
+            }
+        </field>
+    </record>
+
+    <menuitem
+            id="accountant_general_audit_index_a1104_menu"
+            name="Index A.110.4"
+            parent="accountant_general_audit.accountant_risk_assessment_menu"
+            action="accountant_general_audit_index_a1104_action"
+            sequence="4"
```
Comment (andhit-r) [inline accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml]: 1104

**Review by andhit-r — CHANGES_REQUESTED** [review]

File: accountant_general_audit_index_a1104/models/accountant_general_audit_index_a1104_status.py (hunk 12 above)
Comment (azmimr67) [reply accountant_general_audit_index_a1104/models/accountant_general_audit_index_a1104_status.py]: done

File: accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_status_views.xml (hunk 13 above)
Comment (azmimr67) [reply accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_status_views.xml]: done

File: accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_status_views.xml (hunk 14 above)
Comment (azmimr67) [reply accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_status_views.xml:83]: done

File: accountant_general_audit_index_a1104/models/accountant_general_audit_index_a1104.py (hunk 10 above)
Comment (azmimr67) [reply accountant_general_audit_index_a1104/models/accountant_general_audit_index_a1104.py:150]: done

File: accountant_general_audit_index_a1104/models/accountant_general_audit_index_a1104.py (hunk 11 above)
Comment (azmimr67) [reply accountant_general_audit_index_a1104/models/accountant_general_audit_index_a1104.py:216]: done

File: accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_status_views.xml (hunk 14 above)
Comment (azmimr67) [reply accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_status_views.xml:83]: done

File: accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml (hunk 15 above)
Comment (azmimr67) [reply accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml:76]: done

File: accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml (hunk 22 above)
Comment (azmimr67) [reply accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml]: done

File: accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml (hunk 23 above)
Comment (azmimr67) [reply accountant_general_audit_index_a1104/views/accountant_general_audit_index_a1104_views.xml]: done

**Pull request merged**